### DISCUSSION TITLE
V2 모션에서 앉기 애니메이션 스크립트 적용할때 오류

### DISCUSSION THREAD

**Discussion by kimyang-sun** [naverz/zepeto-studio-kor#1230] [5. Q&A]
안녕하세요!
제가 앉기 모션의 애니메이션을 믹사모에서 받아서 적용 후에 V2 적용해주고 동기화까지 해줬습니다.

![image](https://user-images.githubusercontent.com/56675004/186606865-a0e0a2e1-ac4b-4732-aab0-ef5c29c371c6.png)

앉을때 브로드캐스트 할때는 이런식으로 해줬고 SetGesture로 앉는 동작을 적용해준 후에 앉을지점 위치를 받아와서 적용시켰습니다.
(마지막줄은 의자의 자식으로 캐릭터를 넣는 과정입니다.)

여기서 다른사람이 볼 때  V1에서는 그냥 앉는 모션을 강제로 주고 위치를 잡아주니까 잘 됐었는데
V2에서는 해당 위치에 잘 앉을때도 있지만 앉아야 할 위치가 공중이거나 의자와 살짝 겹친다거나 그럴 경우에 
앉는 동작이 적용된 후에 다른 동작들이 막 실행되어서 앉기가 풀려버리는 거 같아요 :(

또 저는 앉을때 index.ts에서 해당 유저가 앉아있는지 여부와 의자 이름을 스키마에 추가해줘서
누군가 새로 접속할때 ClientStarter.ts에서 OnStageChange에 isFirst, OnAddedPlayer.AddListener 안에서
로컬플레이어가 아닐때 SitSync 함수를 실행시켜 다른 앉아있는 사람들의 정보를 받아와 앉아있게 하도록 적용하였습니다. (아래에 코드)

![image](https://user-images.githubusercontent.com/56675004/186608359-bed6cd3d-a7f3-4633-a33f-c77ead31b10c.png)

![image](https://user-images.githubusercontent.com/56675004/186608505-2cc0c72b-e9aa-415f-a0bf-550d7f4c992c.png)

이것도 V1에서는 잘먹혔었는데 V2로 오니까 캐릭터를 받아오면서 캐릭터가 생성될 때 앉는 제스쳐가 실행되지만
이 또한 다른 동작으로 인해 바로 풀려버리는 거 같습니다.

다른 사람의 캐릭터에 앉는 모션을 동기화 해줄 때, V2 모션을 강제로 막는다던가 (앉는 모션이 풀리지 않도록)
마찬가지로 서버에 입장해서 다른 사람의 캐릭터들을 생성해줄때도 앉는 상태 적용할때 V2 모션을 강제로 막거나 하는 방법이
필요할 거 같은데 혹시 방법이 있을까요..?

아니면 앉는 모션이 풀리는 이유가 다른 이유라면 어떤 이유일까요..?

**Comment by zlmrzlo** [comment]
https://github.com/naverz/zepeto-studio-kor/discussions/1218#discussioncomment-3466402
koldshine님의 1번으로 캐릭터의 움직임을 강제로 멈출 수 있을 거에요.
런타임에 생성되는 캐릭터를 직접 활성화/비활성화해보시면 제페토 캐릭터를 어떻게 이용하면 될 것인지 감이 잡힐 것 같네요.

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요,
월드 패키지 1.6.0버전부터 MotionV2 캐릭터 StateMachine Animation을 on/off 할 수 있습니다.
아래 스크립트를 사용해 보세요. 

```typescript
// 더이상 캐릭터의 StateMachine에 의해 애니메이션이 자동재생되지 않습니다.
ZepetoCharacter.StateMachine.constraintStateAnimation = true;

var stateType = UnityEngine.Animator.StringToHash( "State" );
ZepetoCharacter.ZepetoAnimator.SetInteger(stateType, "[Custom Animation State]");
```

**Reply by kimyang-sun** [reply]
답변 정말 감사합니다!!! :)
